src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/RenameStrategies/SimpleRenameStrategy.cs

```diff
@@ -18,7 +19,12 @@ public SimpleRenameStrategy(IProjectThreadingService threadingService, IUserNoti
         }
 
         // For the SimpleRename, it can attempt to handle any situtation
-        public override bool CanHandleRename(string oldFileName, string newFileName) => true;
+        public override bool CanHandleRename(string oldFileName, string newFileName)
+        {
+            var oldNameBase = Path.GetFileNameWithoutExtension(oldFileName);
+            var newNameBase = Path.GetFileNameWithoutExtension(newFileName);
+            return SyntaxFacts.IsValidIdentifier(oldNameBase) && SyntaxFacts.IsValidIdentifier(newNameBase);
+        }
```

Comment:
Thanks for fixing this , can you add few testcases covering this ..

[InlineData("class Foo{}", "Bar.cs", "Foo`.cs")]
[InlineData("class Foo{}", "Bar.cs", "Foo@.cs")]
above test Rename_Symbol_Should_Not_HappenAsync....